### PR TITLE
BREAKING CHANGE: Remove dependency of src/api on src/core, removes multiserver support

### PR DESCRIPTION
**Pull request by clark800**

### CODE REVIEW

**Comment by stevenzeiler** [comment]
boom

File: src/api/transaction/suspended-payment-creation.js [hunk 1]
```diff
@@ -33,10 +32,10 @@ function createSuspendedPaymentCreationTransaction(account: string,
     txJSON.Digest = payment.digest;
   }
   if (payment.allowCancelAfter !== undefined) {
-    txJSON.CancelAfter = utils.fromTimestamp(payment.allowCancelAfter);
+    txJSON.CancelAfter = fromTimestamp(payment.allowCancelAfter);
```
Comment (wltsmrz) [inline src/api/transaction/suspended-payment-creation.js]: Should be named relative to rippled time

File: test/api-test.js [hunk 2]
```diff
@@ -601,8 +600,8 @@ describe('RippleAPI', function() {
     return this.api.getServerInfo().then(() => {
       assert(false, 'Should throw NetworkError');
     }).catch(error => {
-      assert(error instanceof this.api.errors.NetworkError);
-      assert(error.message.indexOf('too much load') !== -1);
+      assert(error instanceof this.api.errors.RippledError);
+      assert(error.message.indexOf('slowDown') !== -1);
```
Comment (wltsmrz) [inline test/api-test.js]: Probably better ways to do this check

**Comment by clark800** [comment]
@wltsmrz updated (i'll squash before merge)

File: src/api/common/connection.js [hunk 3]
```diff
@@ -96,24 +67,26 @@ class Connection extends EventEmitter {
     return this._ws !== null;
   }
 
+  isConnected() {
```
Comment (darkdarkdragon) [inline src/api/common/connection.js]: maybe this should be getter too, as `state`?

File: src/api/ledger/transaction.js [hunk 4]
```diff
@@ -1,28 +1,22 @@
...
-
-function attachTransactionDate(remote: Remote, tx: Object,
-                              callback: CallbackType
-) {
+function attachTransactionDate(connection: Connection, tx: Object
+): Promise<TransactionType> {
   if (tx.date) {
-    callback(null, tx);
-    return;
+    return new Promise(resolve => resolve(tx));
```
Comment (darkdarkdragon) [inline src/api/ledger/transaction.js]: Why not `Promise.resolve(tx)`?

Comment (clark800) [inline src/api/ledger/transaction.js]: Ah, I was looking for that, not sure why I didn't see it in the docs. I'll make that change.

File: src/api/common/connection.js (hunk 3 above)
Comment (clark800) [inline src/api/common/connection.js]: I kind of like the consistency of all public methods being functions; so I'm going to make `state` private and keep this as a function. The function notation makes it more obvious that it's not a settable property.

**Comment by clark800** [comment]
@darkdarkdragon @wltsmrz updated

**Comment by darkdarkdragon** [comment]
LGTM
27 жовт. 2015 19:54 "Chris Clark" <notifications@github.com> пише:

**Comment by clark800** [comment]
Sigh, always forget to squash

**Pull request merged**